crates/ink/codegen/src/generator/metadata.rs

```diff
@@ -338,6 +326,34 @@ impl Metadata<'_> {
         }
     }
 
+    /// Generates ink! metadata for the storage with given selector and indentation.
+    fn generate_constructor_return_type(
+        storage_ident: &Ident,
+        selector_id: u32,
+    ) -> TokenStream2 {
+        let span = storage_ident.span();
+        let constructor_info = quote_spanned!(span =>
+            < #storage_ident as ::ink::reflect::DispatchableConstructorInfo<#selector_id>>
+        );
+        quote_spanned!(span=>
+            ::ink::metadata::ReturnTypeSpec::new(
+                if #constructor_info ::IS_RESULT {
+                    Some(::ink::metadata::TypeSpec::with_name_str::<
+                        ::core::result::Result<
+                            (),
+                            #constructor_info ::Error
+                        >
+                    >(
+                        "core::result::Result"
```

Comment:
I'm not sure of the value of adding this to the display name...really we would want the generic parameters e.g. `Result<(), MyError>`. I know that is slightly tricky but not impossible. The other way would be to just have no display name, anyway the UI should detect whether we are dealing with a result using the type definition.

Anyway, leave it and I can always modify that in my PR